apps/api/src/persistence/migrations/coscrad-query-runner.interface.ts

```diff
@@ -11,6 +12,10 @@ export interface UpdateQueryOptions<TOldDocument> {
  * another db, we would need to abstract over this by forcing the caller
  * to inject the `thin mapping layer` to convert a domain DTO to a database document.
  * When doing this for arango, we convert `id` to `_key`, for example.
+ *
+ * It's questionable whether we really need this abstraction layer. All migrations
```

Comment:
These are those hard lessons. I thought I was doing the right thing because I followed the standard pattern used by ORMs in the Nest ecosystem. I realize now that they have to abstract over the database (PostgreSQL, MySQL, Mongo, etc.), whereas we don't get a lot of benefit in that in the migrations layer (as opposed to the repository pattern, which is great). All migrations are concrete to Arango. A better approach is to remove the formal parameters in the `up \ down` abstraction and use dependency injection for any deps as needed.